staging/src/k8s.io/apiserver/pkg/admission/cel/metrics_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+func TestObserveDefinition(t *testing.T) {
+	defer Metrics.Reset()
+	defer legacyregistry.Reset()
+
+	Metrics.ObserveDefinition(context.TODO(), "active", "deny")
+
+	wantLabels := map[string]string{
+		"state":              "active",
+		"enforcement_action": "deny",
+	}
+
+	expectCounterValue(t, "apiserver_validating_admission_policy_definition_count", wantLabels, 1)
+}
+
+func TestObserveCheck(t *testing.T) {
+	defer Metrics.Reset()
+	defer legacyregistry.Reset()
+
+	Metrics.ObserveCheck(context.TODO(), time.Duration(10)*time.Second, "policy.example.com", "binding.example.com", "true", "allow", "params.example.com", "active")
+	wantLatencyLabels := map[string]string{
+		"policy":                "policy.example.com",
+		"params":                "params.example.com",
+		"policy_binding":        "binding.example.com",
+		"validation_expression": "true",
+		"enforcement_action":    "allow",
+		"state":                 "active",
+	}
+	expectHistogramCountTotal(t, "apiserver_validating_admission_policy_check_duration_seconds", wantLatencyLabels, 1)
+	wantCheckLabels := map[string]string{
+		"policy":                "policy.example.com",
+		"params":                "params.example.com",
+		"policy_binding":        "binding.example.com",
+		"validation_expression": "true",
+		"enforcement_action":    "allow",
+		"state":                 "active",
+	}
+	expectCounterValue(t, "apiserver_validating_admission_policy_check_total", wantCheckLabels, 1)
+}
```

Comment:
I would actually opt for this pattern, it's simpler and you don't need to create an entire utils file for testing metrics. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/metrics_test.go#L32-L159

Reply:
I've removed the utils file and moved to a table test layout with https://github.com/kubernetes/kubernetes/pull/112994/commits/0a65adf525345b0b361bf3695cd961ace02f9a75 - though I've added an extra entry into the table intended to call the right receiver function on `Metrics` (as opposed to calling into the codebase and watching for metrics that way like `endpoints/filters/metrics_test` does), is that alright?

I plan on landing metrics integration with the rest of admission control in a separate PR, if that helps. It shouldn't be too difficult to modify the unit tests to cover the rest of the codebase in that PR, if that's something you think should be done.